### ISSUE TITLE
Arduino IDE reports not a valid library

### ISSUE THREAD

**Issue by G6EJD** [Chocho2017/FireBeetleLEDMatrix#4]
Does not install via the library manager

Examples will not compile, out of date with current IDE versions

**Comment by slovatos** [comment]
Yes, is there any way this can be fixed please??  I really need this to work for a school project and I already tried uploading as a .zip but no luck.  If anyone knows how this can be fixed please let me know!!!

**Comment by G6EJD** [comment]
https://github.com/G6EJD/DFRobot-Firebeetle-LED-Matrix

**Comment by slovatos** [comment]
@G6EJD oh my gosh, thank you so much!! I definitely wasn't expecting a reply this soon, and I really appreciate your help.  One question though, and first let me apologize because I'm still new at using Arduino and github, but the other version doesn't show up on the library manager either and also prompts the same error message: "Specified folder/zip file does not contain a valid library."  Does this mean that I'm doing something wrong?  
Any further clarifications and instructions would be extremely helpful.  Again, thank you very much for all your help and effort.
(For context, the project I'm working on can be found here https://www.instructables.com/Wi-Fi-Smart-Scale-with-ESP8266-Arduino-IDE-Adafrui/ )

**Comment by G6EJD** [comment]
Happy to help.
Get the library from here: https://github.com/Chocho2017/FireBeetleLEDMatrix
Download the zip to your desktop.
Start the IDE, Sketch, manage libraries, add a zip, browse and select it, then it will compile. Check your connections between Firebeetle and display, you can’t plug them together (unfortunately) as they designed the matrix for the older ESP8266 Firebeetle.

**Comment by slovatos** [comment]
Hello again and happy Labor Day! :)  (If you don't celebrate it, then I'll be sure to text you again sometime around Christmas.)
I have tried multiple things to try to add the FireBeetleLEDMatrix library to Arduino, and I think I did it, but I'm not sure.  When I tried adding both Chocho2017's and your .zip files the way you suggested, they provided me with the same error message:
![image](https://user-images.githubusercontent.com/112836471/188542838-360114f2-74f2-4889-a4e0-840cee3ed09d.png)

I tried troubleshooting and following the advice on this website (https://forum.arduino.cc/t/specified-folder-zip-file-does-not-contain-a-valid-library/368616/2), saying to change the name to get rid of hyphens since they aren't valid library characters, but  it still told me the same thing after I did that. D:
So finally, I tried installing the library manually, and I think I did it correctly but I'm not quite sure.  Here's what it says from my end!  Does this look right?
![2022-09-05](https://user-images.githubusercontent.com/112836471/188544296-a36fc205-a3be-497f-a0d5-77d851fb41f6.png)

Regarding your last sentence, what I do have is indeed the older version of the esp8266 which I intend to solder to the LED Matrix so it looks something like this gif:
https://user-images.githubusercontent.com/112836471/188544765-d193f731-9d5c-4ac2-975d-84bfb24f99b6.mp4

...And that's about all I have to say!  Sorry for such a long message.  I know I probably sound like a broken record at this point but again, thank you so so much for the help you've given me.  I see you have a Youtube channel, so I will definitely subscribe & support your awesome hard work any way I can.

**Comment by G6EJD** [comment]
Your library is now correctly installed. If you choose one of the examples or mine, it should all work. The key is the correct wiring and pin assignments in the code. The pins for the ESP8266 will need careful selection to ensure success, you could try:
#define DATA 5
#define CS     4
#define WR    0
#define RD     2

**Comment by slovatos** [comment]
Hello again,
![image](https://user-images.githubusercontent.com/112836471/189502040-821e6fb7-2e58-40e1-8da9-a142d65767c7.png)
I got a sample file to upload in hopes that the lights would work (after I soldered the LED display and the esp8266) but no luck.  I already made sure that I was using the correct board/port, so could it be caused by an issue in the soldering?  I can send a picture/video of what everything else looks like if you want to see!  If its easier to contact you from email or some other messaging system like discord, feel free to let me know. :)

**Comment by G6EJD** [comment]
You’ve got a double definition of your GPIO pins so whether your selection will work will depend on the board type and what you select in the IDE for board type, you need to be very careful as often D0 is not GPIO 0 but some other pin. Also the ESP8266 cannot remap pins whereas the ESP32 can. So you must find and define exactly the pins you’ve used for your board - type?

**Comment by slovatos** [comment]
Okay, so what I noticed was that in the project I'm trying to replicate, a Firebeetle Board ESP8266 Version 1.0 is used, while the one I have is a 2.0... So does that change a lot of things for me?  The next thing I tried was 
#define DATA D6
#define CS D2
#define WR D7
//#define RD D8
#else
#define DATA 5
#define CS 0
#define WR 4
//#define RD 2
So did that make things better or worse? 😂 I feel like I'm getting closer to understanding how to do this right, so I'll count as many small successes as I can get.
My board type for the IDE is the esp 8266, and I used this link: https://raw.githubusercontent.com/DFRobot/FireBeetle-ESP8266/master/package_firebeetle8266_index.json
Maybe it could be outdated?  If you want any pictures to see how it looks from my end let me know.
Thanks again for all your help <3

**Comment by G6EJD** [comment]
Your code is defining the pins then defining them again, so you need to comment out one set.
Your board type needs to be selected as ESP8266 then Firebeetle 8266 you’ll see in down the list of boards could be a way down with the current default of boards in the Arduino IDE. 
if you select the Firebeetle then you can use D0 D4 etc if not then there could be pin misalignment. Search for G6EJD and contact me via my web page for more help

**Comment by slovatos** [comment]
Okay, thank you very much for the clarifications!  Since it is my first time using Arduino I will keep using trial and error until I get it to work.  First I need the display to connect, and then I'll try getting the rest of the pieces of the project to work (Code can be found here: https://content.instructables.com/ORIG/FT9/EE8F/JHKU7D5S/FT9EE8FJHKU7D5S.ino)
Until then, thank you for letting me know about your website (it's really cool! Now I have a way I can support you for all your help!) and if I have any questions I'll shoot you an email.  Feel free to look more at the code in the project to see if it makes sense as well, haha.  Have a good day!

**Comment by slovatos** [comment]
Okay, so what I noticed was that in the project I'm trying to replicate, a
Firebeetle Board ESP8266 Version 1.0 is used, while the one I have is a
2.0... So does that change a lot of things for me?  The next thing I tried
was
#define DATA D6
#define CS D2
#define WR D7
//#define RD D8
#else
#define DATA 5
#define CS 0
#define WR 4
//#define RD 2
So did that make things better or worse? 😂 I feel like I'm getting closer
to understanding how to do this right, so I'll count as many small
successes as I can get.
My board type for the IDE is the esp 8266, and I used this link:
https://raw.githubusercontent.com/DFRobot/FireBeetle-ESP8266/master/package_firebeetle8266_index.json
Maybe it could be outdated?  If you want any pictures to see how it looks
from my end let me know.
Thanks again for all your help <3
